repm/sync_integration_test.go

```diff
@@ -0,0 +1,377 @@
+package repm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	diffserve "roci.dev/diff-server/serve"
+	servetypes "roci.dev/diff-server/serve/types"
+	jsnoms "roci.dev/diff-server/util/noms/json"
+	"roci.dev/replicache-client/db"
+)
+
+// dataLayer is a simple in-proceess data layer. It implements a single
+// mutation 'myPut' which sets a key to a value.
+// dataLayer is NOT safe for concurrent access.
+type dataLayer struct {
+	stores           map[string]*store // keyed by clientID
+	authTokens       map[string]string // keyed by clientID
+	batchServer      *httptest.Server
+	clientViewServer *httptest.Server
+}
+
+// store is the data for a single client.
+type store struct {
+	lastMutationID uint64
+	data           map[string]json.RawMessage
+}
+
+// newDataLayer returns a new dataLayer. Caller must call stop() to clean up.
+func newDataLayer() *dataLayer {
```

Comment:
Nit: prefer literal initialization to `make()`. We should have this be a styleguide thing.